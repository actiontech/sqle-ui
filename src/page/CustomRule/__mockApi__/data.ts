import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
