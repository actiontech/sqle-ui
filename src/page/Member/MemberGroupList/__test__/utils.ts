import { IGetMemberGroupRespDataV1 } from '../../../../api/common';
import user_group from '../../../../api/user_group';
import { resolveThreeSecond } from '../../../../testUtils/mockRequest';

export const mockMemberGroupList: IGetMemberGroupRespDataV1[] = [
  {
    user_group_name: 'test',
    roles: [
      {
        instance_name: 'db1',
        role_names: ['test1', 'test2', 'test3'],
      },
      {
        instance_name: 'db2',
        role_names: ['test1', 'test2', 'test3'],
      },
      {
        instance_name: 'db3',
        role_names: ['test1', 'test2', 'test3'],
      },
      {
        instance_name: 'db4',
        role_names: ['test1', 'test2', 'test3'],
      },
    ],
  },
];

export const mockGetMemberGroups = () => {
  const spy = jest.spyOn(user_group, 'getMemberGroupsV1');
  spy.mockImplementation(() => resolveThreeSecond(mockMemberGroupList));
  return spy;
};

export const mockDeleteMemberGroup = () => {
  const spy = jest.spyOn(user_group, 'deleteMemberGroupV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};
