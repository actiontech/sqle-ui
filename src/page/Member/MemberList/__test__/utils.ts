export const mockList = [
  {
    is_owner: false,
    user_name: 'test',
    roles: [
      {
        instance_name: 'db1',
        role_names: ['test1', 'test2', 'test3'],
      },
      {
        instance_name: 'db2',
        role_names: ['test1', 'test2', 'test3'],
      },
      {
        instance_name: 'db3',
        role_names: ['test1', 'test2', 'test3'],
      },
      {
        instance_name: 'db4',
        role_names: ['test1', 'test2', 'test3'],
      },
    ],
  },
];
