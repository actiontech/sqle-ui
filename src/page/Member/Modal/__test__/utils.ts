import user from '../../../../api/user';
import user_group from '../../../../api/user_group';
import { resolveThreeSecond } from '../../../../testUtils/mockRequest';

export const mockAddMember = () => {
  const spy = jest.spyOn(user, 'addMemberV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockUpdateMember = () => {
  const spy = jest.spyOn(user, 'updateMemberV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockAddMemberGroup = () => {
  const spy = jest.spyOn(user_group, 'addMemberGroupV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockUpdateMemberGroup = () => {
  const spy = jest.spyOn(user_group, 'updateMemberGroupV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};
