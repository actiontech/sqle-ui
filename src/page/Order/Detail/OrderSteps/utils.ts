import { IMaintenanceTimeResV1 } from '../../../../api/common';

export const checkTimeInWithMaintenanceTime = (
  time: moment.Moment,
  maintenanceTime: IMaintenanceTimeResV1[]
) => {
  const hour = time.hour();
  const minute = time.minute();

  if (maintenanceTime.length === 0) {
    return true;
  }

  for (const v of maintenanceTime) {
    const startHour = v.maintenance_start_time?.hour ?? 0;
    const startMinute = v.maintenance_start_time?.minute ?? 0;
    const endHour = v.maintenance_stop_time?.hour ?? 0;
    const endMinute = v.maintenance_stop_time?.minute ?? 0;
    if (startHour === endHour && startHour === hour) {
      if (minute >= startMinute && minute <= endMinute) {
        return true;
      }
    }
    if (hour === startHour) {
      if (minute >= startMinute) {
        return true;
      }
    }
    if (hour === endHour) {
      if (minute <= endMinute) {
        return true;
      }
    }
    if (hour > startHour && hour < endHour) {
      return true;
    }
  }
};
