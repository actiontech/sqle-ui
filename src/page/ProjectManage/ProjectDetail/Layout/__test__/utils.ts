import project from '../../../../../api/project';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetProjectDetail = () => {
  const spy = jest.spyOn(project, 'getProjectDetailV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      create_time: '2022-01-01',
      create_user_name: 'admin',
      desc: 'desc',
      name: 'project1',
    })
  );
  return spy;
};
