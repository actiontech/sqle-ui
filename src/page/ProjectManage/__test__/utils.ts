import { IGetProjectStatisticsResDataV1 } from '../../../api/common';
import project from '../../../api/project';
import statistic from '../../../api/statistic';
import { resolveThreeSecond } from '../../../testUtils/mockRequest';

export const mockGetProjectList = () => {
  const spy = jest.spyOn(project, 'getProjectListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond(
      Array.from({ length: 11 }, (_, i) => ({
        name: `project${i + 1}`,
        desc: 'desc1',
        create_time: '2022-11-01',
        create_user_name: 'admin',
        archived: i < 1,
      })),
      { otherData: { total_nums: 11 } }
    )
  );
  return spy;
};

export const mockDeleteProject = () => {
  const spy = jest.spyOn(project, 'deleteProjectV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockCreateProject = () => {
  const spy = jest.spyOn(project, 'createProjectV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockUpdateProject = () => {
  const spy = jest.spyOn(project, 'updateProjectV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockGetProjectDetail = () => {
  const spy = jest.spyOn(project, 'getProjectDetailV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      create_time: '2022-01-01',
      create_user_name: 'admin',
      desc: 'desc',
      name: 'project1',
      archived: false,
    })
  );
  return spy;
};

export const projectStatisticsData: IGetProjectStatisticsResDataV1 = {
  workflow_total: 12,
  audit_plan_total: 10,
  instance_total: 3,
  member_total: 22,
  rule_template_total: 12,
  whitelist_total: 3,
};

export const mockGetProjectStatistics = () => {
  const spy = jest.spyOn(statistic, 'getProjectStatisticsV1');
  spy.mockImplementation(() => resolveThreeSecond(projectStatisticsData));
  return spy;
};

export const mockArchiveProject = () => {
  const spy = jest.spyOn(project, 'archiveProjectV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockUnarchiveProject = () => {
  const spy = jest.spyOn(project, 'unarchiveProjectV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};
