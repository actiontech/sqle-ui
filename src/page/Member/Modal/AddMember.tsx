import { useBoolean } from 'ahooks';
import { Button, message, Modal } from 'antd';
import { useForm } from 'antd/lib/form/Form';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';
import user from '../../../api/user';
import { IAddMemberV1Params } from '../../../api/user/index.d';
import { ResponseCode } from '../../../data/common';
import EmitterKey from '../../../data/EmitterKey';
import { ModalName } from '../../../data/ModalName';
import { IReduxState } from '../../../store';
import { updateMemberModalStatus } from '../../../store/member';
import EventEmitter from '../../../utils/EventEmitter';
import { useCurrentProjectName } from '../../ProjectManage/ProjectDetail';
import { MemberFormFields } from './index.type';
import MemberForm from './MemberForm';

const AddMember: React.FC = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const { projectName } = useCurrentProjectName();
  const [form] = useForm<MemberFormFields>();
  const [submitLoading, { setFalse: submitFinish, setTrue: startSubmit }] =
    useBoolean();
  const modalVisibility = useSelector(
    (state: IReduxState) => state.member.modalStatus[ModalName.Add_Member]
  );

  const submit = async () => {
    const values = await form.validateFields();
    const params: IAddMemberV1Params = {
      project_name: projectName,
      is_owner: values.isOwner,
      roles: values.roles,
      user_name: values.username,
    };
    startSubmit();
    user
      .addMemberV1(params)
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          message.success(
            t('member.addMember.successTips', {
              name: params.user_name,
            })
          );
          closeModal();
          EventEmitter.emit(EmitterKey.Refresh_Member_List);
          EventEmitter.emit(EmitterKey.Refresh_Filter_User_Tips);
        }
      })
      .finally(() => {
        submitFinish();
      });
  };

  const closeModal = () => {
    form.resetFields();
    dispatch(
      updateMemberModalStatus({
        modalName: ModalName.Add_Member,
        status: false,
      })
    );
  };

  return (
    <Modal
      visible={modalVisibility}
      title={t('member.addMember.modalTitle')}
      closable={false}
      footer={
        <>
          <Button onClick={closeModal} disabled={submitLoading}>
            {t('common.close')}
          </Button>
          <Button type="primary" onClick={submit} loading={submitLoading}>
            {t('common.submit')}
          </Button>
        </>
      }
    >
      <MemberForm form={form} projectName={projectName} />
    </Modal>
  );
};

export default AddMember;
