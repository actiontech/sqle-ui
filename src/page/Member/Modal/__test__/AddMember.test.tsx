import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { useLocation } from 'react-router-dom';
import EmitterKey from '../../../../data/EmitterKey';
import { ModalName } from '../../../../data/ModalName';
import { selectOptionByIndex } from '../../../../testUtils/customQuery';
import {
  mockUseDispatch,
  mockUseSelector,
} from '../../../../testUtils/mockRedux';
import {
  mockUseInstance,
  mockUseRole,
} from '../../../../testUtils/mockRequest';
import EventEmitter from '../../../../utils/EventEmitter';
import AddMember from '../AddMember';
import { mockAddMember } from './utils';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn(),
}));
const projectName = 'test';

describe('test AddMember', () => {
  let addMemberSpy: jest.SpyInstance;
  let dispatchSpy: jest.SpyInstance;
  const emitSpy = jest.spyOn(EventEmitter, 'emit');

  const useLocationMock: jest.Mock = useLocation as jest.Mock;

  beforeEach(() => {
    mockUseRole();
    mockUseInstance();
    addMemberSpy = mockAddMember();
    useLocationMock.mockImplementation(() => {
      return { state: { projectName } };
    });
    dispatchSpy = mockUseDispatch().scopeDispatch;
    mockUseSelector({
      member: {
        modalStatus: {
          [ModalName.Add_Member]: true,
        },
      },
    });
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
    useLocationMock.mockRestore();
  });

  test('should match snapshot', async () => {
    const { baseElement } = render(<AddMember />);
    expect(baseElement).toMatchSnapshot();
  });

  test('should call add member request when clicking submit button', async () => {
    render(<AddMember />);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(addMemberSpy).toBeCalledTimes(0);
    expect(dispatchSpy).toBeCalledTimes(0);
    expect(emitSpy).toBeCalledTimes(0);

    fireEvent.change(screen.getByLabelText('member.memberForm.username'), {
      target: { value: 'name' },
    });
    fireEvent.click(screen.getByLabelText('member.memberForm.projectAdmin'));
    selectOptionByIndex('member.roleSelector.role', 'role_name1');
    selectOptionByIndex('member.roleSelector.instance', 'instance1');

    fireEvent.click(screen.getByText('common.submit'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });

    expect(screen.getByText('common.close').closest('button')).toBeDisabled();
    expect(screen.getByText('common.submit').closest('button')).toHaveClass(
      'ant-btn-loading'
    );

    expect(addMemberSpy).toBeCalledTimes(1);
    expect(addMemberSpy).toBeCalledWith({
      project_name: projectName,
      roles: [{ instance_name: 'instance1', role_names: ['role_name1'] }],
      user_name: 'name',
      is_manager: true,
    });

    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(
      screen.queryByText('member.addMember.successTips')
    ).toBeInTheDocument();
    expect(dispatchSpy).toBeCalledTimes(1);
    expect(dispatchSpy).toBeCalledWith({
      type: 'member/updateModalStatus',
      payload: {
        modalName: ModalName.Add_Member,
        status: false,
      },
    });
    expect(emitSpy).toBeCalledTimes(2);
    expect(emitSpy).toBeCalledWith(EmitterKey.Refresh_Member_List);
    expect(emitSpy).toBeCalledWith(EmitterKey.Refresh_Filter_User_Tips);
    expect(
      screen.getByText('common.close').closest('button')
    ).not.toBeDisabled();
    expect(screen.getByText('common.submit').closest('button')).not.toHaveClass(
      'ant-btn-loading'
    );
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(
      screen.queryByText('member.addMember.successTips')
    ).not.toBeInTheDocument();
  });

  test('should clear form and dispatch "updateMemberModalStatus" when clicking close button', async () => {
    render(<AddMember />);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(dispatchSpy).toBeCalledTimes(0);

    fireEvent.change(screen.getByLabelText('member.memberForm.username'), {
      target: { value: 'name' },
    });

    fireEvent.click(screen.getByText('common.close'));

    expect(screen.getByLabelText('member.memberForm.username')).toHaveValue('');

    expect(dispatchSpy).toBeCalledTimes(1);
    expect(dispatchSpy).toBeCalledWith({
      type: 'member/updateModalStatus',
      payload: {
        modalName: ModalName.Add_Member,
        status: false,
      },
    });
  });
});
