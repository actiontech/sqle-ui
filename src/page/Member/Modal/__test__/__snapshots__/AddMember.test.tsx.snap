// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AddMember should match snapshot 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal"
          role="dialog"
          style="width: 520px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                member.addMember.modalTitle
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal"
              >
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                    >
                      <label
                        class="ant-form-item-required"
                        for="username"
                        title="member.memberForm.username"
                      >
                        member.memberForm.username
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            aria-required="true"
                            class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow ant-select-show-search"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-activedescendant="username_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="username_list"
                                  aria-haspopup="listbox"
                                  aria-owns="username_list"
                                  aria-required="true"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="username"
                                  role="combobox"
                                  type="search"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              >
                                common.form.placeholder.select
                              </span>
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                    >
                      <label
                        class=""
                        for="isManager"
                        title="member.memberForm.projectAdmin"
                      >
                        member.memberForm.projectAdmin
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            aria-checked="false"
                            class="ant-switch"
                            id="isManager"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                    >
                      <label
                        class="ant-form-item-no-colon"
                        title=" "
                      >
                         
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            class="ant-btn ant-btn-dashed ant-btn-block"
                            type="button"
                          >
                            <span
                              aria-label="plus"
                              class="anticon anticon-plus"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="plus"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <defs>
                                  <style />
                                </defs>
                                <path
                                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                                />
                                <path
                                  d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                                />
                              </svg>
                            </span>
                            <span>
                              member.roleSelector.addRole
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn ant-btn-default"
                type="button"
              >
                <span>
                  common.close
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  common.submit
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
